src/modules/Builder/features/ActivityItems/ItemConfiguration/OptionalItemsAndSettings/OptionalItemsAndSettings.hooks.tsx

```diff
@@ -1,119 +1,116 @@
+      // case ItemResponseType.TimeRange:
+      //   return <TimeRange />;
+      // case ItemResponseType.Video:
+      //   return <VideoResponse />;
+      // case ItemResponseType.Photo:
+      //   return <PhotoResponse />;
+      // case ItemResponseType.Date:
+      //   return <Date />;
+      // case ItemResponseType.Audio:
+      //   return <AudioRecord name="audioDuration" />;
       case ItemResponseType.Text:
-        return <TextResponse name="textResponseAnswer" maxCharacters="textResponseMaxCharacters" />;
-      case ItemResponseType.AudioPlayer:
-        return <AudioPlayer name="mediaTranscript" fileResource="mediaFileResource" />;
-      case ItemResponseType.Drawing:
-        return <Drawing drawerImage="drawerImage" drawerBgImage="drawerBgImage" />;
+        return <TextResponse name={name} />;
+      // case ItemResponseType.AudioPlayer:
+      //   return <AudioPlayer name="mediaTranscript" fileResource="mediaFileResource" />;
+      // case ItemResponseType.Drawing:
+      //   return <Drawing drawerImage="drawerImage" drawerBgImage="drawerBgImage" />;
       default:
         null;
     }
-  }, [selectedInputType]);
+  }, [responseType]);
+
+  return activeItem;
+};
 
 export const useSettingsSetup = ({
-  setValue,
-  getValues,
-  watch,
-  register,
-  unregister,
-  clearErrors,
+  name,
   removeOptions,
   handleAddOption,
   removeAlert,
   handleAddAlert,
   setShowColorPalette,
 }: SettingsSetupProps) => {
-  const selectedInputType = watch('itemsInputType');
-  const settings = watch('settings');
-  const selectionRows = watch('selectionRows');
+  const { setValue, getValues, watch } = useFormContext();
 
-  const hasTimer = settings?.includes(ItemConfigurationSettings.HasTimer);
-  const hasAlerts = settings?.includes(ItemConfigurationSettings.HasAlerts);
-  const hasPalette = settings?.includes(ItemConfigurationSettings.HasColorPalette);
-  const isTextInputOptionVisible = settings?.includes(ItemConfigurationSettings.HasTextInput);
-  const isTextInputRequired = settings?.includes(ItemConfigurationSettings.IsTextInputRequired);
-  const isSkippable = settings?.includes(ItemConfigurationSettings.IsSkippable);
+  const settings = watch(`${name}.config`);
 
-  useEffect(() => {
-    setValue('settings', []);
-    setValue('timer', DEFAULT_TIMER_VALUE);
-    removeOptions?.();
-
-    if (
-      selectedInputType === ItemResponseType.SingleSelection ||
-      selectedInputType === ItemResponseType.MultipleSelection
-    ) {
-      handleAddOption?.();
-    }
+  const hasAlerts = get(settings, ItemConfigurationSettings.HasAlerts);
+  const hasPalette = get(settings, ItemConfigurationSettings.HasColorPalette);
+  const isTextInputRequired = get(settings, ItemConfigurationSettings.IsTextInputRequired);
+  const isSkippable = get(settings, ItemConfigurationSettings.IsSkippable);
+
+  const setConfig = (config: Config) => setValue(`${name}.config`, config);
+
+  watch((_, { name: fieldName, type }) => {
```

Comment:
the libarary recommends to unsubscribe from the callback used in watch

Reply:
done